src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/ISnapshotResolver.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+internal interface ISnapshotResolver
+{
+    /// <summary>
+    /// Finds all the projects with a directory that contains the document path. 
+    /// </summary>
+    /// <param name="documentFilePath"></param>
+    /// <param name="includeMiscellaneous">if true, will include the <see cref="GetMiscellaneousProject"/> in the results</param>
+    IEnumerable<IProjectSnapshot> FindPotentialProjects(string documentFilePath, bool includeMiscellaneous);
```

Comment:
Given that all external callers pass `false` for `includeMiscellaneous`, I think we should remove that parameter to make the behaviour of this much more predictable. The implementation of the service can still have it on a private method that does the work, but it seems like a trap for external callers at the moment, and whilst they _currently_ all avoid the trap, it seems unnecessarily fragile.